kubemarine/patches/strong_ciphers_patch.py

```diff
@@ -0,0 +1,113 @@
+
+import yaml
+import time
+from io import StringIO
+from typing import Any, Dict, List
+from textwrap import dedent
+from kubemarine.core.action import Action
+from kubemarine.core.patch import RegularPatch
+from kubemarine.core.resources import DynamicResources
+from kubemarine.core.group import NodeGroup
+
+class TheAction(Action):
+    def __init__(self) -> None:
+        super().__init__("Update API Server TLS cipher suites (if necessary)")
+
+    def run(self, res: DynamicResources) -> None:
+        cluster = res.cluster()
+
+        # Access the raw inventory directly
+        raw_inventory = res.raw_inventory()
+
+        # Check for probable configuration in the initial inventory
+        probable_ciphers_initial = raw_inventory.get("services", {}).get("kubeadm", {}).get("apiServer", {}).get("extraArgs", {}).get("tls-cipher-suites")
+
+        # Compare with desired ciphers
+        desired_ciphers = "TLS_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_RSA_WITH_3DES_EDE_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_256_GCM_SHA384"
+        if probable_ciphers_initial == desired_ciphers:
```

Comment:
Custom ciphers can present and be not equal to the default ones. In this case we have to preserve custom ciphers.

Reply:
Okay now to resolve this in the updated code the defaults ciphers patch will be skipped in case custom ciphers are already present.